### PR TITLE
feat(components/list): be able to disable rows

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

We can't _disable_ a row.

**What is the chosen solution to this problem?**

Add a `getAvaibilityStatus` callback to get the actual status of a row and the deactivation message to show into the tooltip.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/stories/VirtualizedList.js [hunk 1]
```diff
@@ -570,6 +570,33 @@ storiesOf('VirtualizedList', module)
...
+				<br />
+				Here example <pre>{'isDisabled={item => item.id === 6}'}</pre>
+			</p>
+			<IconsProvider defaultIcons={icons} />
+			<section style={{ height: '50vh' }}>
+				<VirtualizedList
+					collection={collection}
+					id={'my-list'}
+					onRowClick={action('onRowClick')}
+					getAvaibilityStatus={item => (item.id === 2 ? { status: false, message: 'Houlala' } : null)}
```
Comment (frassinier) [inline packages/components/stories/VirtualizedList.js]: What about 
```suggestion
					getUnavaibilityMessage={row => (row.id === 2 ? { message: 'Houlala' } : null)}
```
which means that it not available

File: packages/components/stories/VirtualizedList.js [hunk 2]
```diff
@@ -570,6 +570,33 @@ storiesOf('VirtualizedList', module)
...
 		</div>
 	))
+	.add('List > Table : desactivation', () => (
+		<div className="virtualized-list">
+			<h1>Virtualized List</h1>
+			<p>
+				Row can be disabled by passing <b>isDisabled</b> function
+				that returns if a row is disabled.
+				<br />
+				Here example <pre>{'isDisabled={item => item.id === 6}'}</pre>
```
Comment (frassinier) [inline packages/components/stories/VirtualizedList.js]: Description looks wrong

Comment (ncomont) [reply packages/components/stories/VirtualizedList.js]: Ho yes, thanks

**Comment by jsomsanith-tlnd** [comment]
LGTM but can you add a story please ?

**Pull request merged**